demos/common/python/openvino/model_zoo/model_api/models/instance_segmentation.py

```diff
@@ -0,0 +1,293 @@
+ limitations under the License.
+"""
+
+import cv2
+import numpy as np
+
+from .image_model import ImageModel
+from .utils import RESIZE_TYPES, load_labels, nms, pad_image
+
+
+class MaskRCNNModel(ImageModel):
+    def __init__(self, model_adapter, prob_threshold=0.5, labels=None, keep_aspect_ratio=False):
+        super().__init__(model_adapter)
+        self.resize_type = 'fit_to_window' if keep_aspect_ratio else 'standard'
+        self.resize = RESIZE_TYPES[self.resize_type]
+        self.prob_threshold = prob_threshold
+        self._check_io_number((1, 2), (3, 4, 5, 8))
+        self.type = 'segmentoly' if len(self.inputs) == 2 else 'mask_rcnn'
+        if isinstance(labels, (list, tuple)):
+            self.labels = labels
+        else:
+            self.labels = load_labels(labels) if labels else None
+
+        self.output_blob_name = self._get_outputs()
+
+    def _get_outputs(self):
+        if self.type == 'segmentoly':
+            return self._get_segmentoly_outputs()
+        outputs = {}
+        for layer_name in self.outputs:
+            if layer_name.startswith('TopK'):
+                continue
+            layer_shape = self.outputs[layer_name].shape
+
+            if len(layer_shape) == 1:
+                outputs['labels'] = layer_name
+            elif len(layer_shape) == 2:
+                outputs['boxes'] = layer_name
+            elif len(layer_shape) == 3:
+                outputs['masks'] = layer_name
+            else:
+                raise Exception("Unexpected output layer shape {} with name {}".format(layer_shape, layer_name))
+
+        return outputs
+
+    def _get_segmentoly_outputs(self):
+        outputs = {}
+        for layer_name in self.outputs:
+            layer_shape = self.outputs[layer_name].shape
+            if layer_name == 'boxes' and len(layer_shape) == 2:
+                outputs['boxes'] = layer_name
+            elif layer_name == 'classes' and len(layer_shape) == 1:
+                outputs['labels'] = layer_name
+            elif layer_name == 'scores' and len(layer_shape) == 1:
+                outputs['scores'] = layer_name
+            elif layer_name == 'raw_masks' and len(layer_shape) == 4:
+                outputs['masks'] = layer_name
+            else:
+                raise Exception("Unexpected output layer shape {} with name {}".format(layer_shape, layer_name))
+        return outputs
+
+    def preprocess(self, inputs):
+        image = inputs
+        meta = {'original_shape': image.shape}
+        resized_image = self.resize(image, (self.w, self.h))
+        meta.update({'resized_shape': resized_image.shape})
+        input_image_size = resized_image.shape[:2]
+        if self.resize_type == 'fit_to_window':
+            resized_image = pad_image(resized_image, (self.w, self.h))
+        resized_image = self.input_transform(resized_image)
+        resized_image = self._change_layout(resized_image)
+        dict_inputs = {self.image_blob_name: resized_image}
```

Comment:
```suggestion
        dict_inputs, meta = super().preprocess(inputs)
        input_image_size = meta['resized_shape'].shape[:2]
```
Since the implementation of `MaskRCNNModel` `preprocess` repeats the common steps implemented in `ImageModel` method